runtime/compiler/trj9/control/RecompilationInfo.hpp

```diff
@@ -372,6 +390,8 @@ class TR_PersistentJittedBodyInfo
 
    bool       isLongRunningInterpreted() const { return _longRunningInterpreted; }
 
+   void setProfileInfo(TR_PersistentProfileInfo * ppi) { _profileInfo = ppi; }
+   TR_PersistentProfileInfo *getProfileInfo() { return _profileInfo; }
```

Comment:
Isn't it possible that if someone acquires a reference to the bodyInfo indirectly (ie via the J9JITExceptionTable) and then decides to dereference the profileInfo, because `getProfileInfo` doesn't go through the process of acquiring the lock (and thus doing a ref count inc), they could try and access freed memory? If so, it is worth adding a comment here to state that `getProfileInfo` should only be called via the `TR_PersistentMethodInfo` or `J9::Recompilation` objects, as they will do the proper work of locking and incrementing the ref count.

Reply:
Yes, you are right. These methods rely on the profile information persisting at least until `jitReleaseCodeCollectMetaData` cleans up the meta data for a body, setting this pointer to `NULL` in the process. Uses of these calls can all guarantee that they are safe, as they will remain within this duration, such as when the body is actively being compiled. I'll add a comment documenting this behaviour.